### PR TITLE
BYO LLM 8.16 updates

### LINKED ISSUE #5965: [Request] Update setup local LLM page
### Description

We introduced a new enhancement of the local LLM setup where we add a new `Other (OpenAI Compatible Service)` provider in the OpenAI connector. This will clarify configuration steps to the user and will allow to avoid ambiguity in which option to use for local LLMs.

### Background & resources

* PRs: https://github.com/elastic/kibana/pull/194831
* Issues/metas: https://github.com/elastic/security-team/issues/8936
* Point of contact: @e40pud 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

8.16

### Serverless release

The week of October 21st, 2024

### Feature differences

The feature is identical in ESS and Serverless.

### API docs impact

We need to update this page https://www.elastic.co/guide/en/security/current/connect-to-byo-llm.html#_configure_the_connector_in_your_elastic_deployment and mention that user should select `Other (OpenAI Compatible Service)` as OpenAI provider. Let me create a doc ticket for that.

cc @leemthompo 

### Prerequisites, privileges, feature flags

_No response_

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5965 — updates the bring your own LLM guide to reflect a change in the Elastic Integration configuration interface for OpenAI connectors. 

Previews: [BYO LLM ](https://elastic-dot-co-docs-production-af2c81ztl-elastic-dev.vercel.app/current/serverless/security/connect-to-byo-llm)(serverless version)

### CODE REVIEW

**Review by e40pud — DISMISSED** [review]
Thank you, this looks great!

cc @jamesspi

**Comment by jamesspi** [comment]
LGTM 👍 

FYI @Charelzard

File: docs/AI-for-security/connect-to-byo.asciidoc [hunk 1]
```diff
@@ -180,10 +180,11 @@ Finally, configure the connector:
 1. Log in to your Elastic deployment.
 2. Navigate to **Stack Management → Connectors → Create Connector → OpenAI**. The OpenAI connector enables this use case because LM Studio uses the OpenAI SDK.
 3. Name your connector to help keep track of the model version you are using.
-4. Under **URL**, enter the domain name specified in your Nginx configuration file, followed by `/v1/chat/completions`.
-5. Under **Default model**, enter `local-model`.
-6. Under **API key**, enter the secret token specified in your Nginx configuration file.
-7. Click **Save**.
+4. Under **Select an OpenAI provider** select **Other (OpenAI Compatible Service)**.
```
Comment (nastasha-solomon) [inline docs/AI-for-security/connect-to-byo.asciidoc:183]:
```suggestion
4. Under **Select an OpenAI provider**, select **Other (OpenAI Compatible Service)**.
```

File: docs/serverless/AI-for-security/connect-to-byo-llm.mdx [hunk 2]
```diff
@@ -160,10 +160,11 @@ Finally, configure the connector in your Security project:
 1. Log in to your Security project.
 2. Navigate to **Stack Management → Connectors → Create Connector → OpenAI**. The OpenAI connector enables this use case because LM Studio uses the OpenAI SDK.
 3. Name your connector to help keep track of the model version you are using.
-4. Under **URL**, enter the domain name specified in your Nginx configuration file, followed by `/v1/chat/completions`.
-5. Under **Default model**, enter `local-model`.
-6. Under **API key**, enter the secret token specified in your Nginx configuration file.
-7. Click **Save**.
+4. Under **Select an OpenAI provider** select **Other (OpenAI Compatible Service)**.
```
Comment (nastasha-solomon) [inline docs/serverless/AI-for-security/connect-to-byo-llm.mdx:163]: Sorry, I should've left a comment here too. 

```suggestion
4. Under **Select an OpenAI provider**, select **Other (OpenAI Compatible Service)**.
```

**Review by nastasha-solomon — APPROVED** [review]

**Review by e40pud — APPROVED** [review]

**Pull request merged**